xdp.go

```diff
@@ -380,11 +380,16 @@ func NewSocket(Ifindex int, QueueID int, options *SocketOptions) (xsk *Socket, e
 		return nil, fmt.Errorf("syscall.Bind SockaddrXDP failed: %v", err)
 	}
 
-	xsk.freeDescs = make([]bool, options.NumFrames)
-	for i := 0; i < options.NumFrames; i++ {
-		xsk.freeDescs[i] = true
+	xsk.freeRXDescs = make([]bool, options.NumFrames)
+	xsk.freeTXDescs = make([]bool, options.NumFrames)
+	for i := range xsk.freeRXDescs {
+		xsk.freeRXDescs[i] = true
 	}
-	xsk.getDescs = make([]Desc, 0, options.NumFrames)
+	for i := range xsk.freeTXDescs {
+		xsk.freeTXDescs[i] = true
+	}
+	xsk.getTXDescs = make([]Desc, 0, options.CompletionRingNumDescs)
+	xsk.getRXDescs = make([]Desc, 0, options.FillRingNumDescs)
```

Comment:
This change to use FillRing / CompletionRing when creating the desc slices is interesting.

Still trying to reason about what it changes -- for instance, should `NumFrames == CompletionRingNumDescs + FillRingNumDescs` / or are they different? Do we still need a separate NumFrames option, ... ? Need to review AF_XDP docs.

Reply:
right, this PR is based on assumption that NumFrames == CompletionRingNumDescs + FillRingNumDescs; if NumFrames < CompletionRingNumDescs + FillRingNumDescs, then I don't think it gona work since NumFrames is the max number of available decs for both TX and RX; on theory  NumFrames could > CompletionRingNumDescs + FillRingNumDescs, but I don't see the use case of that